src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -234,6 +234,19 @@ internal BoundExpression BindRValueWithoutTargetType(ExpressionSyntax node, Diag
             return BindToNaturalType(BindValue(node, diagnostics, BindValueKind.RValue), diagnostics, reportDefaultMissingType);
         }
 
+        internal BoundExpression BindToTypeForErrorRecovery(BoundExpression expression, TypeSymbol type = null)
+        {
+            if (expression is null)
+                return null;
+            var discardedDiagnostics = DiagnosticBag.GetInstance();
+            var result =
+                (!expression.NeedsToBeConverted() || expression.WasConverted) ? expression :
+                type is null ? BindToNaturalType(expression, discardedDiagnostics) :
```

Comment:
>BindToNaturalType(expression, discardedDiagnostics) [](start = 31, length = 51)

Consider passing ```reportDefaultMissingType=false```. Even though that will not have any observable effect outside of the compiler, this will decrease unnecessary allocations of diagnostic.

Reply:
This is error recovery code.  As such it is executed rarely, and maintainability is more important than micro-optimizations.

---
In reply to: [329687441](https://github.com/dotnet/roslyn/pull/38750#discussion_r329687441) [](ancestors = 329687441)

Reply:
> This is error recovery code.  As such it is executed rarely, and maintainability is more important than micro-optimizations.

Could you elaborate please on the maintainability issue that you are referring to. What is it. Why it is a problem, etc? 

---
In reply to: [330260351](https://github.com/dotnet/roslyn/pull/38750#discussion_r330260351) [](ancestors = 330260351,329687441)

Reply:
Generally speaking, less code, fewer parameters, etc, is easier to maintain because the reader has less to understand while maintaining the code.  Code that is more complex than necessary is a problem because it is more complex to maintain.

---
In reply to: [330262709](https://github.com/dotnet/roslyn/pull/38750#discussion_r330262709) [](ancestors = 330262709,330260351,329687441)

Reply:
I do not believe that adding an argument in this particular case creates a maintainability problem. I'll add it myself after this PR gets merged.

---
In reply to: [330264141](https://github.com/dotnet/roslyn/pull/38750#discussion_r330264141) [](ancestors = 330264141,330262709,330260351,329687441)